dcos/marathon.py

```diff
@@ -54,7 +54,11 @@ def _to_error(response):
     if isinstance(response, Error):
         return DefaultError(_default_marathon_error(response.error()))
 
-    message = response.json().get('message')
+    try:
+        message = response.json().get('message')
+    except ValueError as e:
```

Comment:
OK, I understand now.  We should first check the return code, and only try to parse the 'message' if it's a 404.  This 400 should result in the Bad Request message.  See `http._default_to_exception` for an example